src/Headless/Avalonia.Headless/HeadlessUnitTestSession.cs

```diff
@@ -137,9 +140,12 @@ private IDisposable EnsureApplication()
 
         return Disposable.Create(() =>
         {
-            scope.Dispose();
+            ((ToolTipService?)AvaloniaLocator.Current.GetService<IToolTipService>())?.Dispose();
+            (AvaloniaLocator.Current.GetService<FontManager>() as IDisposable)?.Dispose();
             Dispatcher.ResetForUnitTests();
-            SynchronizationContext.SetSynchronizationContext(null);
+            scope.Dispose();
+            Dispatcher.ResetBeforeUnitTests();
+            SynchronizationContext.SetSynchronizationContext(oldContext);
```

Comment:
Also reordered operations here to match https://github.com/AvaloniaUI/Avalonia/blob/22f3c026dfb16f68808235cee01176f590ca4cf8/tests/Avalonia.UnitTests/UnitTestApplication.cs#L40-L61